### PR TITLE
Add support for exportable entity statuses

### PR DESCRIPTION
**Pull request by earlyburg**
Entities in Backdrop do not have statuses by default, however some projects ported from Drupal 7 use this functionality.

This fixes some issues with search_api and another module that I am porting, Search API Solr
In D7 exportable entities had statuses that reflected how these entities were created. Search API module uses these in:

https://github.com/backdrop-contrib/search_api/blob/1.x-1.x/search_api.admin.inc
(lines 69, 94, 119, 515, 1126)

and 

https://github.com/backdrop-contrib/search_api/blob/1.x-1.x/search_api.module
(line 2681)

After this is discussed and pushed we could theroretically make a pull request in the search_api contrib project to change this:

$row[] = theme('entity_status', array('status' => $server->status));

to this:

$row[] = theme('entity_plus_status', array('status' => $server->status));

thus enhancing the exportable entity functioning, fixing errors and enabling more functionality in the Search API suite of tools.

Thanks for your time I appreciate it.

### CODE REVIEW

**Comment by laryn** [comment]
Thanks @earlyburg! @hosef (the other maintainer) is planning to look at this next week and do a code review.

File: entity_plus.module [hunk 1]
```diff
@@ -4,6 +4,43 @@
  * Returns a property wrapper for given data.
  */
 
+ /**
+  * Defines status codes used for exportable entities.
+  */
+
+ /**
+  * A bit flag used to let us know if an entity is in the database.
+  */
```
Comment (hosef) [inline entity_plus.module:13]: I don't think that this comment here is necessary.

File: entity_plus.module [hunk 2]
```diff
@@ -912,6 +949,10 @@ function entity_plus_theme() {
   $pattern = '(' . implode('|', $types) . ')(\.|__)';
 
   return array(
+		'entity_plus_status' => array(
```
Comment (hosef) [inline entity_plus.module]: A few of the code sections have tabs at the start instead of spaces. Could you please fix those.

**Review by hosef — COMMENTED** [review]
There are just 2 minor things that I think should be tidied up. Other than that, the changes look fine to me.

File: entity_plus.module (hunk 1 above)
Comment (earlyburg) [reply entity_plus.module:13]: Agree, I'll remove it.

File: entity_plus.module (hunk 2 above)
Comment (earlyburg) [reply entity_plus.module]: My Bad there. I will make these updates and let you know asap. Thank you, I really appreciate it.

**Comment by earlyburg** [comment]
I've completed the code cleanup. Thanks for your patience.

**Comment by laryn** [comment]
Whoops, did you merge this directly @earlyburg? That is a maintainer task.

**Comment by earlyburg** [comment]
Sorry. I did not know that. I misunderstood what was happening.

**Comment by laryn** [comment]
@earlyburg Not to worry in this case, I think. I'll let @hosef double check things when he's back in a few days.

I think we're slowly transitioning off of a more permissive configuration now that Github allows more complex team permissions, but some projects aren't there yet and we're still running off the "don't merge if you're not the maintainer" handshake agreement ([see the first point in the contributor agreement](https://github.com/backdrop-ops/contrib#backdrop-contributed-project-agreement)). ☺️

**Comment by earlyburg** [comment]
I apologize for causing trouble for you. Thank you for taking the time to explain it to me. From now on I will be more team focused.

**Comment by laryn** [comment]
@earlyburg 
No trouble -- glad you're here!

**Comment by hosef** [comment]
@laryn I took another look at the changes, and everything looks fine to me.

**Pull request merged**